### PR TITLE
Migrate cluster resource controller to call admin service endpoints

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
To support scenarios where cluster resource controller is deployed on a separate cluster without flyteadmin database credentials, this change moves all db call to flyteadmin service calls.

A big part of this change was refactoring execution cluster target logic so that a separate component is used to list **all** execution cluster targets. This is now independent of the GetTarget() logic which depends on database access to fetch matchable attributes overrides.

Tested on flyte sandbox by deleting a project quota, having the updated cluster resource controller run, and then asserting the project quota got re-added.

Note, that we'll still need to keep the execution cluster config as part of deploying the cluster resource controller, but if it's deployed on the same cluster as the flytepropeller cluster, the in target cluster provider will suffice.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/2059

## Follow-up issue
_NA_

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -44,6 +44,7 @@ require (
 	github.com/spf13/cobra v1.1.3
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.0
+	github.com/zalando/go-keyring v0.1.1
```
Comment (pmahindrakar-oss) [inline go.mod:47]: I remember having this discussion with @EngHabu  where we didn't want to add this dependency in stdlib and keep it only in flytectl where we used the keyring 
Now that we have flyteadmin also needing this dependency, is it worth moving this to stdlib and also Token implemetation

File: cmd/entrypoints/util/token_cache_keyring.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package pkce
+
```
Comment (pmahindrakar-oss) [inline cmd/entrypoints/util/token_cache_keyring.go:2]: May be we can move this from flytectl to stdlib so that admin can consume aswell?

File: go.mod (hunk 1 above)
Comment (katrogan) [reply go.mod:47]: no longer relevant - marking resolved

File: cmd/entrypoints/util/token_cache_keyring.go (hunk 2 above)
Comment (katrogan) [reply cmd/entrypoints/util/token_cache_keyring.go:2]: thanks for the callout, removed!

File: cmd/entrypoints/clusterresource.go [hunk 3]
```diff
@@ -3,71 +3,66 @@ package entrypoints
...
 	Short: "This command administers the ClusterResourceController. Please choose a subcommand.",
 }
 
-func GetLocalDbConfig() repositoryConfig.DbConfig {
-	return repositoryConfig.DbConfig{
-		Host:   "localhost",
-		Port:   5432,
-		DbName: "postgres",
-		User:   "postgres",
+func getClusterResourceController(ctx context.Context) clusterresource.Controller {
```
[the remark this answers is not in the document]
Comment (katrogan) [reply cmd/entrypoints/clusterresource.go]: sounds good, done

**Comment by katrogan** [comment]
PTAL @EngHabu @pmahindrakar-oss

File: cmd/entrypoints/clusterresource.go [hunk 4]
```diff
@@ -3,32 +3,50 @@ package entrypoints
...
+	if err != nil {
+		panic(err)
+	}
+
+	clientSet, err := admin.ClientSetBuilder().WithConfig(admin.GetConfig(ctx)).Build(ctx)
+	if err != nil {
+		panic(err)
 	}
+
+	return clusterresource.NewClusterResourceController(clientSet.AdminClient(), listTargetsProvider, scope)
```
Comment (EngHabu) [inline cmd/entrypoints/clusterresource.go:49]: Will it crash if it can't talk to admin? just making sure it'll... specially for enabling auth in admin, this needs to know to detect that and crash...

File: cmd/entrypoints/util/token_cache_keyring.go (hunk 2 above)
Comment (EngHabu) [reply cmd/entrypoints/util/token_cache_keyring.go:2]: Still see this...

**Review by EngHabu — APPROVED** [review]

File: cmd/entrypoints/clusterresource.go (hunk 4 above)
Comment (katrogan) [reply cmd/entrypoints/clusterresource.go:49]: it will dump a bunch of error logs complaining admin is unreachable and fail

File: cmd/entrypoints/util/token_cache_keyring.go (hunk 2 above)
Comment (katrogan) [reply cmd/entrypoints/util/token_cache_keyring.go]: sorry, removed the references to it but forgot to delete. mind taking another look?

**Review by EngHabu — APPROVED** [review]
